CMSgov/easi-app
# [EASI-2113] Refactor cedar contact select

# EASI-2113

## Changes and Description

- Refactors `<CedarContactSelect>` (work from [EASI-1685](https://jiraent.cms.gov/browse/EASI-1685)) to use React Select library
  - EASI-2113 has modifications to the select that weren't possible with previous library used (Reach combobox)
  - Removes Reach Combobox as dependency
- This does not address EASI-2113 acceptance criteria - will be done in additional PR once this is merged



## How to test this change

Check acceptance criteria from [EASI-1685](https://jiraent.cms.gov/browse/EASI-1685) to make sure no bugs were introduced when refactoring

1. Start new system intake - field is on Contact Details page
2. Search for business owner name
3. Ensure selecting business owner name sets email field
4. Test "Business owner same as requester" checkbox
6. Set product owner and isso
7. Test additional contacts form (hidden behind feature flag)
8. Complete system intake - contacts should show up correctly on review screen at end of form



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
